RELEASE-NOTES.md

```diff
@@ -48,6 +48,7 @@
 * Added log into the file with loaded external plugins
 * Warn users if they attempt to use templating with project definition version 1
 * Improved output and format of Pydantic validation errors
+* The `snow app run` command will no longer set `debug_mode=true` on upgrade unless explicitly set in `snowflake.yml`
```

Comment:
nit: maybe something like

```
The `snow app run` command will no longer override the debug mode on during an application upgrade unless [...]
```

Just to make it clear what "debug_mode=true" means in case a customer encountered that behaviour in the past (which is somewhat likely I'd bet)